New functionality to share with Facebook and Twitter selectively

vick08 opened the pull request:
Introduced the new method called "shareA" that allows sharing with Facebook or Twitter selectively (using SLComposeViewController). "shareA" takes four parameters:
1.  Service Name (Facebook, Twitter or Sinaweibo). -- string
2.  Text to tweet -- string.
3.  URL to post -- string.
4.  Path to the image to attach -- string.

A call would look something like this:

<code>
window.plugin.social.shareA('Facebook', 'I love guitar!', 'http://www.victortsaran.net', 'my image.gif');
</code>

I am willing to see this merged into the main "share" method but I didn't want to do anything major until I hear back from the original author.

Looking forward to your feedback.
Victor

Commit:
Added shareA function that allows posting to Facebook and Twitter separately.

vick08 commented:
Cameron.
Do you have any plans looking at my pull request? I am curious what you think and whether you find it valuable. If yes, I'd love to hear suggestions on how this can be improved, etc.

nebrekab commented:
@vick08 not sure if this helps, but you can achieve that by creating a standard mailto link in your phonegap application, leaving out  the email address. eg: ahref="mailto:?subject=Check this out!&body=http://www.github.com". I used this technique and via JS updated the share link in the href link. Obviously this link sits separately to the social share triggering link.

@bfcam @vick08  I have noticed that if the user is NOT signed into Facebook or Twitter BUT DOES have it installed on their phone then pressing the respective icon to share fails silently. Is there any way to detect if Facebook or Twitter apps are running, thus enabling sharing? 

Oh, and thanks, these plugins are excellent! ;)

vick08 commented:
@nebrekab, the "mailto" approach will work fine, however, you will not be able to provide attachments in your email. Or, so I think… :)
The "silent die" issue: I saw that if no Twitter or Facebook are configured, you will be displayed a standard iOS dialog asking you to configure them. However, have not tested the case where you are not signed in. Interestingly enough, I have not seen to query the "log in" status from the "social" framework. Will have a look when I get a chance.
Thanks for bringing that up.

nebrekab commented:
@vick08 Thanks for the reply.

Yeah, good point about attachments, that won't allow for it out-of-the-box. Although there's a work around here: http://stackoverflow.com/questions/7046100/mailto-link-with-attachment - not tried it out, but what they are saying seems to make sense if you are trying to link to a file in an email triggered fro ma mailto link.

After a quick Google search I came across this: http://stackoverflow.com/questions/12705559/in-ios-6-check-if-a-person-is-logged-in-to-a-social-network - looks like in Objective-C you can check the login status, although I know very little about Objective-C! :)

andremalkine commented:
re: Silent Fail - I found that removing the conditional at the end of shareA in Social.m

```
if([SLComposeViewController isAvailableForServiceType:serviceType]) { ... }
```

prevents the script from simply aborting when you try to launch the unconfigured service. I think the trend of checking for service availability tends to pervade around Social.framework, but people are ultimately going to try to use it to share content by any means possible. Users are not going to assume it's their own fault that an option is missing or inert.

**edit:** that is, comment out the conditional, without removing the code inside it.

vick08 committed:
- Removed the "service availability" conditional for now to force iOS to
display its own configuration dialog if the requested service is not
configured. This is a temporary fix until I figure out something better.

vick08 commented:
@andremalkine,
Thanks for your suggestion. Since I could not find anything better, I removed the "service availability" conditional for now.
Thanks again!

originalp commented:
Great addition, however, it crashes when the image parameter is not entered so I leave it blank like this:
window.plugins.social.shareA('Twitter','message', 'http://url.com', ' ');

vick08 commented:
@bfcam, was curious if you would be interested in accepting this pull request or you feel this doesn't belong in your plugin?
